components/experimental/src/transliterate/transliterator/mod.rs

```diff
@@ -306,21 +306,31 @@ impl Transliterator {
                 // 1. Insert a placeholder to avoid infinite recursion.
                 env.insert(dep.to_string(), InternalTransliterator::Null);
                 // 2. Load the transliterator, by checking
-                let internal_t = dep
-                    // a) hardcoded specials
-                    .strip_prefix("x-")
-                    .map(|special| Transliterator::load_special(special, normalizer_provider))
-                    // b) the user-provided override
-                    .or_else(|| Transliterator::load_with_override(&dep, lookup?).transpose())
-                    // c) the data
-                    .unwrap_or_else(|| {
+                let mut internal_t = None;
+                // a) hardcoded specials
+                if let Some(special) = dep.strip_prefix("x-") {
+                    internal_t = Transliterator::load_special(special, normalizer_provider)?;
+                }
+                // b) the user-provided override
+                if let Some(lookup) = lookup {
+                    if internal_t.is_none() {
+                        if let Ok(locale) = dep.parse() {
+                            internal_t = Transliterator::load_with_override(&locale, lookup)?;
+                        }
+                    }
+                }
+                // c) the data
+                let internal_t = match internal_t {
+                    Some(t) => t,
+                    None => {
                         let rbt = Transliterator::load_rbt(
                             #[allow(clippy::unwrap_used)] // infallible
                             DataMarkerAttributes::try_from_str(&dep.to_ascii_lowercase()).unwrap(),
                             transliterator_provider,
                         )?;
-                        Ok(InternalTransliterator::RuleBased(rbt))
-                    })?;
```

Comment:
I find this much more readable than the mutable `Option` you're using.

Reply:
Problems I had with the previous code:

1. It was concealing the bug
2. The functions involved return a mix of Option and Result and it is hard to keep track of which level you are at in unwrapping the values
3. It's difficult to run the debugger with all the closures. You have to dip in and out of the Rust standard library with all the closures.

But I can convert this back to using the option chain if you prefer, as the code owner.